streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,254 @@
+                streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG), time);
+
+    }
+
+
+    /**
+     * Cretes a new topic with the given number of partitions and replication factor.
+     *
+     * @param internalTopicConfig
+     * @param numPartitions
+     * @param replicationFactor
+     */
+    public void createTopic(final InternalTopicConfig internalTopicConfig, final int numPartitions, final int replicationFactor, final long windowChangeLogAdditionalRetention) {
+        Properties topicProperties = internalTopicConfig.toProperties(windowChangeLogAdditionalRetention);
+        Map<String, String> topicConfig = new HashMap<>();
+        for (String key : topicProperties.stringPropertyNames()) {
+            topicConfig.put(key, topicProperties.getProperty(key));
+        }
+        CreateTopicsRequest.TopicDetails topicDetails = new CreateTopicsRequest.TopicDetails(numPartitions, (short) replicationFactor, topicConfig);
+        Map<String, CreateTopicsRequest.TopicDetails> topics = new HashMap<>();
+        topics.put(internalTopicConfig.name(), topicDetails);
+
+        CreateTopicsRequest createTopicsRequest = new CreateTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+
+        ClientResponse clientResponse = sendRequest(createTopicsRequest.toStruct(), ApiKeys.CREATE_TOPICS, null);
+        CreateTopicsResponse createTopicsResponse = new CreateTopicsResponse(clientResponse.responseBody());
+        if (createTopicsResponse.errors().get(internalTopicConfig.name()).code() > 0) {
+            throw new StreamsException("Could not create topic: " + internalTopicConfig.name() + ". " + createTopicsResponse.errors().get(internalTopicConfig.name()).name());
+        }
+    }
+
+    /**
+     * Delete a topic.
+     *
+     * @param topic
+     */
+    public void deleteTopic(final String topic) {
+        log.debug("Deleting topic {} from ZK in partition assignor.", topic);
+        Set<String> topics = new HashSet();
+        topics.add(topic);
+
+        DeleteTopicsRequest deleteTopicsRequest = new DeleteTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+
+        ClientResponse clientResponse = sendRequest(deleteTopicsRequest.toStruct(), ApiKeys.DELETE_TOPICS, null);
+        DeleteTopicsResponse deleteTopicsResponse = new DeleteTopicsResponse(clientResponse.responseBody());
+        if (deleteTopicsResponse.errors().get(topic).code() > 0) {
+            throw new StreamsException("Could not delete topic: " + topic + ". " + deleteTopicsResponse.errors().get(topic).name());
+        }
+
+    }
+
+    /**
+     * Send a request to kafka broker of this client. Polls the request for a given number of iterations to receive the response.
+     *
+     * @param request
+     * @param apiKeys
+     * @param callback
+     */
+    public ClientResponse sendRequest(final Struct request, final ApiKeys apiKeys, final RequestCompletionHandler callback) {
```

Comment:
private? is only used in this class, so there is no need to make it public.
Also, as per comment below, we can probably remove the `callback` arg as we don't really need it, i.e., we can just pass `null` for the callback.